pkg/images/enricher/enricher_impl.go

```diff
@@ -711,11 +715,13 @@ func filterRegistriesBySource(requestSource *RequestSource, registries []registr
 }
 
 func getMatchingRegistries(registries []registryTypes.ImageRegistry,
-	image *storage.Image) ([]registryTypes.ImageRegistry, error) {
-	var matchingRegistries []registryTypes.ImageRegistry
+	image *storage.Image) (map[string][]registryTypes.ImageRegistry, error) {
+	matchingRegistries := make(map[string][]registryTypes.ImageRegistry, len(image.GetNames().GetNames()))
 	for _, registry := range registries {
-		if registry.Match(image.GetName()) {
-			matchingRegistries = append(matchingRegistries, registry)
+		for _, name := range image.GetNames().GetNames() {
+			if registry.Match(image.GetName()) {
```

Comment:
Don't you mean
```suggestion
			if registry.Match(name.GetName()) {
```
here?